benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/LocalWorker.java

```diff
@@ -338,10 +338,12 @@ public void internalMessageReceived(int size, long publishTimestamp) {
         bytesReceivedCounter.add(size);
 
         // NOTE: PublishTimestamp is expected to be using the wall-clock time across
-        // machines
+        // machines in milliseocnds
         Instant currentTime = Instant.now();
+
         long currentTimeNanos = TimeUnit.SECONDS.toNanos(currentTime.getEpochSecond()) + currentTime.getNano();
-        long endToEndLatencyMicros = TimeUnit.NANOSECONDS.toMicros(currentTimeNanos - publishTimestamp);
+        long publishTimeNanos = TimeUnit.MILLISECONDS.toNanos(publishTimestamp);
```

Comment:
So `publishTimestamp` was in milliseconds?

Reply:
This seems is pretty low resolution for a latency which is often on the order of a few milliseconds total.

Reply:
Yeah, this is a major flaw with the E2E latency measuring across all of the OMB providers - because we're using the Kafka message timestamp, we're only sending it in milliseconds. 
Ideally we'd fix this across the board and use a message header rather than the message timestamp, but we'd then need to fix OMB everywhere

Reply:
We have swarm workload which measure in true ns.

Reply:
Updated